File: src/main/java/com/group/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.java
package com.group.libraryapp.domain.user.loanhistory;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserLoanHistoryRepository extends JpaRepository<UserLoanHistory, Long> {

	UserLoanHistory findByBookNameAndIsReturn(String bookName, boolean isReturn);

	List<UserLoanHistory> findByUserName(String userName);
}

File: src/main/kotlin/com/group/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.kt
package com.group.libraryapp.domain.user.loanhistory

import org.springframework.data.jpa.repository.JpaRepository

interface UserLoanHistoryRepository : JpaRepository<UserLoanHistory, Long> {

    fun findByBookNameAndStatus(bookName: String, status: UserLoanStatus): UserLoanHistory?

    fun findByUserName(userName: String): List<UserLoanHistory>

    fun findAllByStatus(status: UserLoanStatus): List<UserLoanHistory>

}
